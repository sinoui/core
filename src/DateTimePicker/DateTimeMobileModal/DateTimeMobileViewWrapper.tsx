import styled from 'styled-components';
import Paper from '@sinoui/core/Paper';

const disabledProps = ['in'];
const DateTimeMobileViewWrapper = styled(Paper).withConfig({
  shouldForwardProp: (prop) => !disabledProps.includes(prop),
})<any>`
  width: 328px;

  .sinoui-week-title-bar {
    padding: 0 12px;
  }

  .sinoui-date-time-mobile-view__datesview {
    padding: 0 12px;
    height: 240px;
  }
`;

export default DateTimeMobileViewWrapper;
