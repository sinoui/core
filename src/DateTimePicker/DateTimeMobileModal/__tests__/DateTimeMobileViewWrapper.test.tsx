import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { defaultTheme } from '@sinoui/theme';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import DateTimeMobileViewWrapper from '../DateTimeMobileViewWrapper';

it('渲染日期时间选择视图容器', () => {
  const { getByTestId } = render(
    <ThemeProvider theme={defaultTheme}>
      <DateTimeMobileViewWrapper data-testid="wrapper" />
    </ThemeProvider>,
  );

  const wrapper = getByTestId('wrapper');

  expect(wrapper).toHaveStyleRule('width', '328px');
  expect(wrapper).toHaveStyleRule('padding', '0 12px', {
    modifier: '.sinoui-week-title-bar',
  });
  expect(wrapper).toHaveStyleRule('padding', '0 12px', {
    modifier: '.sinoui-date-time-mobile-view__datesview',
  });
  expect(wrapper).toHaveStyleRule('height', '240px', {
    modifier: '.sinoui-date-time-mobile-view__datesview',
  });
});

it('镜像测试', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <DateTimeMobileViewWrapper data-testid="wrapper" />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
