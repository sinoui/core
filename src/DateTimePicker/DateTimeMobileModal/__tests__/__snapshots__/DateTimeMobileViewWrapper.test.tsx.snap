// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`镜像测试 1`] = `
.c0 {
  box-sizing: border-box;
  border-radius: 4px;
  background-color: #fff;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  -webkit-transition: box-shadow 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: box-shadow 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
}

.c1 {
  width: 328px;
}

.c1 .sinoui-week-title-bar {
  padding: 0 12px;
}

.c1 .sinoui-date-time-mobile-view__datesview {
  padding: 0 12px;
  height: 240px;
}

<div
  className="c0 c1 sinoui-paper"
  data-testid="wrapper"
/>
`;
